### PR TITLE
[2201.5.0-stage] Fix character class with multiple items being incorrectly populated

### LINKED ISSUE #40141: [Bug]: Multiple range character class patterns defined in runtime crashes
### Description

```ballerina
import ballerina/lang.regexp;

public function main() returns error? {
    string:RegExp pattern = check regexp:fromString("[a-zA-Z]");
    _ = pattern.find("Ballerina");
}
```

```
error: java.lang.ClassCastException {"message":"class io.ballerina.runtime.internal.values.RegExpCharSet cannot be cast to class java.lang.String (io.ballerina.runtime.internal.values.RegExpCharSet is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')"}
        at ballerina.lang.regexp.0:findImpl(regexp.bal:81)
           ballerina.lang.regexp.0:find(regexp.bal:71)
           regexp_bbe:main(regexp_bbe.bal:5)
```

### Steps to Reproduce

_No response_

### Affected Version(s)

2201.5.0 rc1

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

Fixes #40141

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/langlib-test/src/test/resources/test-src/regexp_test.bal [hunk 1]
```diff
@@ -1266,6 +1266,71 @@ function testRuntimeRegexpParser() {
...
+    assertEquality("1", (<regexp:Span>res6).substring());
+    
+    string:RegExp pattern7 = check regexp:fromString("[a-z.\\s\\w]");
+    assertTrue(pattern7 is string:RegExp);
+    if (pattern7 is string:RegExp) {
+        assertTrue(re `[a-z.\s\w]` == pattern7);
+    }
+    regexp:Span? res7 = pattern7.find("abc");
+    assertTrue(res7 is regexp:Span);
+    assertEquality("a", (<regexp:Span>res7).substring());
```
Comment (SasinduDilshara) [inline langlib/langlib-test/src/test/resources/test-src/regexp_test.bal:1331]: Shall we add a test for #40136 using `fromString()`

Comment (pcnfernando) [reply langlib/langlib-test/src/test/resources/test-src/regexp_test.bal:1331]: It won't go in this path

**Review by SasinduDilshara — APPROVED** [review]

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/regexp/TreeBuilder.java [hunk 2]
```diff
@@ -239,24 +240,26 @@ private RegExpCharSet readRegCharSet() {
...
             if (reCharSet.getCharSetAtoms().length > 0) {
-                charSetAtoms.add(reCharSet);
+                charSetAtoms.addAll(Arrays.asList(reCharSet.getCharSetAtoms()));
             }
-
             return new RegExpCharSet(charSetAtoms.toArray());
         }
         RegExpCharSet reCharSetNoDash = readCharSetNoDash(nextToken);
-        return new RegExpCharSet(new Object[]{startReCharSetAtom, reCharSetNoDash});
+        charSetAtoms.add(startReCharSetAtom);
```
Comment (pcnfernando) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/regexp/TreeBuilder.java:258]: Can we move the common logic out?

**Review by pcnfernando — APPROVED** [review]
Tested. LGTM

**Pull request merged**